### DISCUSSION TITLE
A tree-sitter grammar for Soufflé

### DISCUSSION THREAD

**Discussion by langston-barrett** [souffle-lang/souffle#2333] [Show and tell]
I've written a fairly complete tree-sitter grammar for Soufflé. It successfully parses 500/503 test cases from the Soufflé repo (v2.3) and the entirety of ddisasm. Tree-sitter generates a pure C parser with bindings to a variety of languages.

https://github.com/langston-barrett/tree-sitter-souffle

**Comment by luc-tielen** [comment]
@langston-barrett maybe also interesting to look at integration for tree-sitter powered editors? e.g. for neovim there is: https://github.com/nvim-treesitter/nvim-treesitter.

**Reply by langston-barrett** [reply]
@luc-tielen Happy to accept PRs! https://github.com/langston-barrett/tree-sitter-souffle/issues/16
